docs/source/package_reference/utilities.mdx

```diff
@@ -60,3 +60,54 @@ See below for all custom errors thrown by different methods across the package.
 [[autodoc]] huggingface_hub.utils.EntryNotFoundError
 
 [[autodoc]] huggingface_hub.utils.LocalEntryNotFoundError
+
+## Validators
+
+`huggingface_hub` include custom validators to validate method arguments automatically.
+Validation is inspired by the work done in [Pydantic](https://pydantic-docs.helpmanual.io/)
+to validate type hints but with more limited features.
+
+### Generic decorator
+
+[`~utils.validate_hf_hub_args`] is a generic decorator to encapsulate
+methods that have arguments following `huggingface_hub`'s naming. By default, all
+arguments that has a validator implemented will be validated.
+
+If an input is not valid, a [`~utils.HFValidationError`] is thrown. Only
+the first non-valid value throws an error and stops the validation process.
+
+Usage:
+
+```py
+>>> from huggingface_hub.utils import validate_hf_hub_args
+
+>>> @validate_hf_hub_args
```

Comment:
Very very nice decorator, this should be super useful around the codebase and in third-party libraries.